swap/protocol_test.go

```diff
@@ -432,7 +432,10 @@ loop:
 		case <-ctx.Done():
 			t.Fatal("Expected one cheque, but there is none")
 		default:
-			if creditor.getLastSentCheque() != nil {
+			creditor.lock.Lock()
```

Comment:
hm. this is test code, but the production version is still not acquiring the swap peer mutex as the getter documentation describes:
```go
// PeerCheques returns the last sent and received cheques for a given peer
func (s *Swap) PeerCheques(peer enode.ID) (PeerCheques, error) {
	var pendingCheque, sentCheque, receivedCheque *Cheque

	swapPeer := s.getPeer(peer)
	if swapPeer != nil {
		pendingCheque = swapPeer.getPendingCheque()
		sentCheque = swapPeer.getLastSentCheque()
		receivedCheque = swapPeer.getLastReceivedCheque()
	}
```

I suggest to sweep through all getter methods that expect a lock held to find similar mistakes.
And once again this is why I don't like having such getters in the codebase in the first place. I've argued against this already when the initial SWAP PR was being under review. We should eliminate the _possibilities_ for similar developer mistakes by having a simple design.

Reply:
I added the missing locks, all peer access should now be under lock (except in `NewPeer` and `AddPeer` where it is clear that no other go routine could have a reference at this point).

We will most likely get rid of the getters and setters soon as they also don't play nice with the batch writes we want to use in some places. I will discuss this with the team in our next call.